src/quicr_client_raw_session.h

```diff
@@ -0,0 +1,251 @@
+
+  /**
+   * @brief Stop subscription on the given QUICR namespace
+   *
+   * @param quicr_namespace       : Identifies QUICR namespace
+   * @param origin_url            : Origin serving the QUICR Session
+   * @param auth_token            : Auth Token to validate the Subscribe
+   *                                Request
+   */
+  void unsubscribe(const quicr::Namespace& quicr_namespace,
+                   const std::string& origin_url,
+                   const std::string& auth_token) override;
+
+  /**
+   * @brief Publish Named object
+   *
+   * @param quicr_name               : Identifies the QUICR Name for the object
+   * @param priority                 : Identifies the relative priority of the
+   *                                   current object
+   * @param expiry_age_ms            : Time hint for the object to be in cache
+   *                                      before being purged after reception
+   * @param use_reliable_transport   : Indicates the preference for the object's
+   *                                   transport, if forwarded.
+   * @param data                     : Opaque payload
+   *
+   */
+  void publishNamedObject(const quicr::Name& quicr_name,
+                          uint8_t priority,
+                          uint16_t expiry_age_ms,
+                          bool use_reliable_transport,
+                          bytes&& data) override;
+
+  /**
+   * @brief Publish Named object
+   *
+   * @param quicr_name               : Identifies the QUICR Name for the object
+   * @param priority                 : Identifies the relative priority of the
+   *                                   current object
+   * @param expiry_age_ms            : Time hint for the object to be in cache
+                                       before being purged after reception
+   * @param use_reliable_transport   : Indicates the preference for the object's
+   *                                   transport, if forwarded.
+   * @param offset                   : Current fragment offset
+   * @param is_last_fragment         : Indicates if the current fragment is the
+   * @param data                     : Opaque payload of the fragment
+   */
+  void publishNamedObjectFragment(const quicr::Name& quicr_name,
+                                  uint8_t priority,
+                                  uint16_t expiry_age_ms,
+                                  bool use_reliable_transport,
+                                  const uint64_t& offset,
+                                  bool is_last_fragment,
+                                  bytes&& data) override;
+
+  void handle(messages::MessageBuffer&& msg);
+  void removeSubscribeState(bool all, const quicr::Namespace& quicr_namespace,
+                            const SubscribeResult::SubscribeStatus& reason);
+
+  std::shared_ptr<qtransport::ITransport> transport;
+  qtransport::LogHandler& log_handler;
+
+protected:
+  std::mutex mutex;
```

Comment:
mutex has a variable name is confusing, can we please rename it to idetify its purpose ?